actions/trigger-argo-workflow/cmd/trigger-argo-workflow/argo.go

```diff
@@ -0,0 +1,151 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log/slog"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/cenkalti/backoff/v4"
+)
+
+type App struct {
+	levelVar *slog.LevelVar
+	logger   *slog.Logger
+
+	argoToken string
+
+	addCILabels bool
+
+	command          string
+	workflowTemplate string
+	extraArgs        []string
+
+	namespace string
+	instance  string
+
+	retries uint64
+}
+
+func (a App) server() string {
+	instanceToHost := map[string]string{
+		"dev":  "argo-workflows-dev",
+		"ops": "argo-workflows",
+	}
+	return fmt.Sprintf("%s.grafana.net:443", instanceToHost[a.instance])
+}
+
+func (a App) env() []string {
+	// Argo CLI requires either `Token ` or `Bearer ` before the token, add it if it's missing.
+	if !strings.HasPrefix(a.argoToken, "Bearer") && !strings.HasPrefix(a.argoToken, "Token") {
+		a.argoToken = "Bearer " + a.argoToken
+	}
+
+	return []string{
+		"ARGO_HTTP1=true",
+		"KUBECONFIG=/dev/null",
+		fmt.Sprintf("ARGO_NAMESPACE=%s", a.namespace),
+		fmt.Sprintf("ARGO_SERVER=%s", a.server()),
+		fmt.Sprintf("ARGO_TOKEN=%s", a.argoToken),
+	}
+}
+
+var nameRe = regexp.MustCompile(`^Name:\s+(.+)`)
+
+func (a App) outputWithURI(input *bytes.Buffer) (string, string) {
+	output := input.String()
+
+	matches := nameRe.FindStringSubmatch(output)
+
+	var uri string
+	if len(matches) == 2 {
+		uri = fmt.Sprintf("https://%s/workflows/%s/%s", a.server(), a.namespace, matches[1])
```

Comment:
What happens in case that's not true? Should we throw an error and fail?

Reply:
You won't get a URL in the output from the workflow, but it only matters when the `argo` command managed to run successfully (otherwise _that_ would cause the program to fail).

At the time I didn't think that should cause a fatal error - make the GHA run fail - but do you think differently? Would be happy to change it.

Reply:
Ah cool, that's only for the output. Then no need to cause a fatal error as you say, however it'd may be a good solution to throw a warning or something, like "couldn't get the URI" so we can know where to look in case there's no output. But no strong opinions on that.

Reply:
Alrighty, let me add a warning, good idea

Reply:
Fixed in d7bc9812733d7b981fa9207a81b15e55faf64a2a!